services/google-data-api/src/index.js

```diff
@@ -0,0 +1,83 @@
+const { service } = require('@base-cms/micro');
+const { stringify } = require('querystring');
+
+const { name, version } = require('../package.json');
+const fetch = require('./fetch');
+const {
+  connect,
+  ping,
+  retrieve,
+  write,
+} = require('./mongodb');
+
+const { log } = console;
+
+const sortObject = (object = {}) => {
+  const keys = Object.keys(object).sort();
+  return keys.reduce((obj, key) => ({ ...obj, [key]: object[key] }), {});
+};
+
+module.exports = service.json({
+  init: async () => {
+    log(`> Booting ${name} ${version}...`);
+    log('> Connecting to MongoDB...');
+    await connect();
+  },
+  ping,
+  actions: {
+    /**
+     * The Youtube Channel List API
+     * @see https://developers.google.com/youtube/v3/docs/channels/list
+     */
+    'youtube.channelList': async (params) => {
+      const uri = 'https://www.googleapis.com/youtube/v3/channels';
+      const {
```

Comment:
Move to top params arg and ensure it has a default object val, e.g. `{ param1, param2 } = {}`